_posts/2016-12-27-angular-2-advance-testing-with-custom-matchers.md

```diff
@@ -0,0 +1,650 @@
+---
+layout: post
+title: Angular Testing Directives with Custom Matchers
+date: 2016-12-27T00:00:00.000Z
+update_date: 2016-12-27T00:00:00.000Z
+imageUrl: /images/banner/testing-directives-with-custom-matchers.jpg
+summary: >-
+  Deliver terse, DRY, self-documenting Karma tests with Angular using custom Jasmine Matchers and 
+  Helper functions.
+categories:
+  - angular
+tags:
+  - angular2
+  - karma
+  - jasmine
+  - testing directives
+  - custom matchers
+  - typescript
+  - special helper functions
+  - DRY
+topic: testings
+author: thomas_burleson
+related_posts:
+  - Testing Services with Http in Angular
+related_videos:
+  - '193524896'
+
+---
+
+This is not another article on basic Jasmine/Karma testing with Angular 2. Rather this is a more 
+advanced article on DRY Karma testing techinques using **Custom Matchers** and **Special Helpers**... 
+these techniques will make your unit-tests incredibly easy to read and to maintain.
+
+To achieve our goals, there are three (3) important testing topics that I wanto to cover:
+
+* Testing custom Angular **Directives**
+* Building reusable, DRY TestBed **Helper** methods
+* Using Typescript Jasmine **Custom Matchers**
+
+These techniques are practically undocumented... yet they are also fundamentally critical to writing 
+quality tests.
+
+To whet your appetite, here are some sample DRY tests that I will be showing you how to write:
+
+![pic1](https://cloud.githubusercontent.com/assets/210413/21514513/8a9690ea-cc8b-11e6-98ac-2a66a42f023d.png)
+
+![pic2](https://cloud.githubusercontent.com/assets/210413/21514536/bf6531dc-cc8b-11e6-8d5f-fad4a1d82be9.png)
```

Comment:
Great pictures with annotations!

Can you add those to the `/images` folder and then grab them from there? We want the blog to work offline with all resources as well :)